rcl/src/rcl/expand_topic_name.c

```diff
@@ -167,20 +167,14 @@ rcl_expand_topic_name(
           *output_topic_name = NULL;
           char * unmatched_substitution =
             rcutils_strndup(next_opening_brace, substitution_substr_len, allocator);
-          char * allocated_msg = NULL;
-          char * msg = NULL;
           if (unmatched_substitution) {
-            allocated_msg = rcutils_format_string(
+            RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
               allocator,
               "unknown substitution: %s", unmatched_substitution);
-            msg = allocated_msg;
           } else {
-            SAFE_FWRITE_TO_STDERR("failed to allocate memory for error message\n");
-            msg = "unknown substitution: allocation failed when reporting error";
+            SAFE_FWRITE_TO_STDERR("failed to allocate memory for unmatched substitution\n");
```

Comment:
I think this macro name changed.

Reply:
I'd be fine with it using the one from `rcutils`, the reason it did not before is probably because it was not in a public header.

Reply:
Do you want to change that, or move this pr forward as-is?